oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/remote/client/RemoteClientManager.java

```diff
@@ -230,7 +246,10 @@ private void reBuildRemoteClients(List<RemoteInstance> remoteInstances) {
 
         remoteClientCollection.values()
                               .stream()
-                              .filter(remoteClientAction -> remoteClientAction.getAction().equals(Action.Close))
+                              .filter(remoteClientAction ->
+                                          remoteClientAction.getAction().equals(Action.Close)
+                                              && !remoteClientAction.getRemoteClient().getAddress().isSelf()
```

Comment:
```suggestion
                                              && 
remoteClientAction.getRemoteClient() != null
                                              &&                                           !remoteClientAction.getRemoteClient().getAddress().isSelf()
```

Reply:
This would not be null. I enhanced another by following `Filter OAP instances(unassigned in booting stage) of the empty IP in KubernetesCoordinator`.